docs/docs/learn/programming/mcp.md

```diff
@@ -0,0 +1,187 @@
+---
+sidebar_position: 3
+---
+
+# Model Context Protocol (MCP)
+
+The [Model Context Protocol (MCP)](https://modelcontextprotocol.io/) is an open protocol that standardizes how applications provide context to language models. DSPy supports MCP, allowing you to use tools from any MCP server with DSPy agents.
+
+## Installation
+
+Install DSPy with MCP support:
+
+```bash
+pip install -U "dspy[mcp]"
+```
+
+## Overview
+
+MCP enables you to:
+
+- **Use standardized tools** - Connect to any MCP-compatible server
+- **Share tools across stacks** - Use the same tools across different frameworks
+- **Simplify integration** - Convert MCP tools to DSPy tools with one line
+
+As long as you have an `mcp.ClientSession`, you can use MCP tools with DSPy's `ReAct` module or any other tool-based workflow.
+
+## Using MCP with DSPy
+
+### 1. Stdio Server (Local Process)
+
+The most common way to use MCP is with a local server process communicating via stdio:
```

Comment:
this is a question not comment - do you think HTTP connection is more commonly used than Stdio now? Stdio is a strange usage especially in production environment IMO